src/periphery/GmxGlpRewardHandler.sol

```diff
@@ -0,0 +1,177 @@
+
+        emit LogFeeParametersChanged(_feeCollector, _feePercent);
+    }
+
+    /// @param token The allowed reward tokens to swap
+    function setRewardTokenEnabled(IERC20 token, bool enabled) external onlyOwner {
+        rewardTokenEnabled[token] = enabled;
+        emit LogRewardTokenUpdated(token, enabled);
+    }
+
+    /// @param token The allowed token out support when swapping rewards
+    function setSwappingTokenOutEnabled(IERC20 token, bool enabled) external onlyOwner {
+        swappingTokenOutEnabled[token] = enabled;
+        emit LogSwappingTokenOutUpdated(token, enabled);
+    }
+
+    /// @param recipient Allowed recipient for token out when swapping
+    function setAllowedSwappingRecipient(address recipient, bool enabled) external onlyOwner {
+        allowedSwappingRecipient[recipient] = enabled;
+        emit LogAllowedSwappingRecipientUpdated(recipient, enabled);
+    }
+
+    function setRewardRouter(IGmxRewardRouterV2 _rewardRouter) external onlyOwner {
+        emit LogRewardRouterChanged(rewardRouter, _rewardRouter);
+        rewardRouter = _rewardRouter;
+    }
+
+    function setSwapper(address _swapper) external onlyOwner {
+        emit LogSwapperChanged(swapper, _swapper);
+        swapper = _swapper;
+    }
+
+    ///////////////////////////////////////////////////////////////////////
+    // GMX/esGMX Handling
+    // Thanks to RageTrade
+
+    /// @notice unstakes and vest protocol esGmx to convert it to Gmx
+    function unstakeAndVestEsGmx(uint256 amount) external onlyOwner {
+        // unstakes the protocol esGMX and starts vesting it
+        // this encumbers some glp deposits
+        // can stop vesting to enable glp withdraws
+        rewardRouter.unstakeEsGmx(amount);
+        IGmxVester(rewardRouter.glpVester()).deposit(amount);
+    }
+
+    /// @notice claims vested gmx tokens (i.e. stops vesting esGmx so that the relevant glp amount is unlocked)
+    /// @dev when esGmx is vested some GlP tokens are locked on a pro-rata basis, in case that leads to issue in withdrawal this function can be called
+    function stopVestAndStakeEsGmx() external onlyOwner {
+        // stops vesting and stakes the remaining esGMX
+        // this enables glp withdraws
+        IGmxVester(rewardRouter.glpVester()).withdraw();
+        uint256 esGmxWithdrawn = IERC20(rewardRouter.esGmx()).balanceOf(address(this));
+        rewardRouter.stakeEsGmx(esGmxWithdrawn);
+    }
+
+    /// @notice claims vested gmx tokens to feeRecipient
+    /// @dev vested esGmx gets converted to GMX every second, so whatever amount is vested gets claimed
+    function claimVestedGmx() external onlyOwner {
+        // stops vesting and stakes the remaining esGMX
+        // this can be used in case glp withdraws are hampered
+        uint256 gmxClaimed = IGmxVester(rewardRouter.glpVester()).claim();
+
+        //Transfer all of the gmx received to fee recipient
+        IERC20(rewardRouter.gmx()).safeTransfer(feeCollector, gmxClaimed);
```

Comment:
maybe instead of transferring it to the fee distributor it could stay here to increase earnings?
we want to use the multiplier if possible.

Reply:
the handleRewards already stake and compound to increase the multiplier etc. This call should when we want to exit.